2996_reflection/reflection.md

```diff
@@ -1113,6 +1137,31 @@ constexpr auto struct_to_tuple(T const& t) {
 :::
 
 
+### Syntax discussion
+
+Early discussions of splice-like constructs (related to the TS design) considered using `unreflexpr(...)` for that purpose.
+P1240R0 adopted that option for _expression_ splicing, observing that a single splicing syntax could not viably be parsed (some disambiguation is needed to distinguish types and templates).
+SG-7 eventually agreed with the `[: ... :]` syntax --- with disambiguating tokens such as `typename` where needed --- which is a little lighter and more distinctive.
+
+We propose `[:` and `:]` be single tokens rather than combinations of `[`, `]`, and `:`.
+Among others, it simplifies the handling of expressions like `arr[[:refl():]]`.
+On the flip side, it requires a special rule like the one that was made to handle `<::` to leave the meaning of `arr[::N]` unchanged.
+
+A syntax that is delimited on the left and right is useful here because spliced expressions may involve lower-precedence operators.
+However, there are other possibilities.
+For example, now that `@$@` is available in the basic source character set, we might consider `@$@<@_expr_@>`.
```

Comment:
Alright I'm going to merge this (after I fix some formatting and typos) but we're seriously proposing syntax that adds angle brackets right? Surely `${e}` and `$(e)` are both way better than `$<e>` right?

Notably tho, some some template systems (ahem CMake) do use `@` for substitution. But everyone hates CMake syntax, so I'm not sure anybody would _really_ argue that using `@` for reflecting is inconsistent with CMake. 

Another potentially good use for `$` would be a placeholder for `|>` [P2672](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2672r0.html) which given that nobody has found the time to work on traits or the brilliant insight to make them work, maybe worth revisiting.